src/Compilers/CSharp/Test/Semantic/Semantics/CollectionLiteralTests.cs

```diff
@@ -15,290 +15,1566 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+                //         if (args.Length > 0) throw [];
+                Diagnostic(ErrorCode.ERR_CollectionLiteralTargetTypeNotConstructible, "[]").WithArguments("System.Exception").WithLocation(5, 36),
+                // (6,15): error CS0029: Cannot implicitly convert type 'System.Collections.Generic.List<int>' to 'System.Exception'
+                //         throw [1, 2, 3];
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "[1, 2, 3]").WithArguments("System.Collections.Generic.List<int>", "System.Exception").WithLocation(6, 15));
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void NaturalType_38()
+        {
+            string source = """
+                using System;
+                class Program
+                {
+                    static void Main()
+                    {
+                        Console.WriteLine([1, 2, 3] is []);
+                        Console.WriteLine([1, 2, 3] is [_, _, _]);
+                    }
+                }
+                """;
+            CompileAndVerify(source, targetFramework: TargetFramework.Net70, expectedOutput: """
+                False
+                True
+                """);
+        }
+
+        [Fact]
+        public void NaturalType_39()
+        {
+            string source = """
+                struct S { }
+                class Program
+                {
+                    static void Main()
+                    {
+                        S s = [null, Unknown];
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (6,22): error CS0103: The name 'Unknown' does not exist in the current context
+                //         S s = [null, Unknown];
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "Unknown").WithArguments("Unknown").WithLocation(6, 22));
+        }
+
+        [Fact]
+        public void OverloadResolution_01()
+        {
+            string source = """
+                using System.Collections.Generic;
+                class Program
+                {
+                    static IEnumerable<int> F1(IEnumerable<int> arg) => arg;
+                    static int[] F1(int[] arg) => arg;
+                    static int[] F2(int[] arg) => arg;
+                    static IEnumerable<int> F2(IEnumerable<int> arg) => arg;
+                    static void Main()
+                    {
+                        var x = F1([]);
+                        var y = F2([1, 2]);
+                        x.Report(includeType: true);
+                        y.Report(includeType: true);
+                    }
+                }
+                """;
+            // PROTOTYPE: Should these calls be ambiguous when we treat IEnumerable<T> as a valid target type (since it is an interface of List<T>)?
```

Comment:
`int[]` implements `IEnumerable<int>`, so I expect that no, these would not be ambiguous.

Reply:
If `int[]` and `IEnumerable` are both considered constructible collection types, should _better function member_ prefer one over the other?

Reply:
i def think we would want a preference.  having an array and an IEnumerable overload would be pretty normal.  Having that be ambiguous woudl be unfortunate.